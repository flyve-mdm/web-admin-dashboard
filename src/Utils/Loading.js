
import React, { Component } from 'react'
import PropTypes from 'prop-types'

class Loading extends Component {
    render () {
        const headerSize = this.props.headerSize ? this.props.headerSize : 0

        return (
            <div className="loading" style={{ marginTop: - headerSize, display: 'flex', height: '100%', justifyContent: 'center', textAlign: 'center', alignItems: 'center', flexDirection: 'column' }}>
                <div>
                    <div className='loader'>
                        <div className='circle'></div>
                        <div className='circle'></div>
                        <div className='circle'></div>
                        <div className='circle'></div>
                        <div className='circle'></div>
                    </div>   
                    <p>{this.props.message}</p>
                </div>
            </div>
        )
    }
}

Loading.propTypes = {
    message: PropTypes.string,
    headerSize: PropTypes.number
}

export default Loading