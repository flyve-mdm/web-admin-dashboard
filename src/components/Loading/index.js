/*
*   Copyright © 2018 Teclib. All rights reserved.
*
*   This file is part of web-mdm-dashboard
*
* web-mdm-dashboard is a subproject of Flyve MDM. Flyve MDM is a mobile
* device management software.
*
* Flyve MDM is free software: you can redistribute it and/or
* modify it under the terms of the GNU General Public License
* as published by the Free Software Foundation; either version 3
* of the License, or (at your option) any later version.
*
* Flyve MDM is distributed in the hope that it will be useful,
* but WITHOUT ANY WARRANTY; without even the implied warranty of
* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
* GNU General Public License for more details.
* ------------------------------------------------------------------------------
* @author     Gianfranco Manganiello (gmanganiello@teclib.com)
* @author     Hector Rondon (hrondon@teclib.com)
* @copyright  Copyright © 2018 Teclib. All rights reserved.
* @license    GPLv3 https://www.gnu.org/licenses/gpl-3.0.html
* @link       https://github.com/flyve-mdm/web-mdm-dashboard
* @link       http://flyve.org/web-mdm-dashboard
* @link       https://flyve-mdm.com
* ------------------------------------------------------------------------------
*/

import React from 'react'
import PropTypes from 'prop-types'

/**
 * Component of loading
 * @param {object} props
 * @return {component} loading component
 */
const Loading = props => {
    /** Create loader */
    const loader = (
        <div className='loader'>
            <div className='circle'></div>
            <div className='circle'></div>
            <div className='circle'></div>
            <div className='circle'></div>
            <div className='circle'></div>
        </div>  
    )

    /** Create a small or normal load component, according to what was requested */
    const loadComponent = props.small ? (
        <div className="loading loading--small" style={props.style}>
            <div> { loader } </div>
        </div>
    ) : (
        <div className="loading" style={{marginTop: - props.headerSize, ...props.style}}>
            <div>
                { loader }
                <p>{props.message}</p>
            </div>
        </div>
    )

    /** Render component */
    return loadComponent
}

Loading.defaultProps = {
    style: {},
    small: false,
    headerSize: 0,
    message: ''
}

Loading.propTypes = {
    message: PropTypes.string,
    headerSize: PropTypes.number,
    small: PropTypes.bool,
    style: PropTypes.object
}

export default Loading