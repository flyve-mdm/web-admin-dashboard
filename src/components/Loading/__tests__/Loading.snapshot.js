import React from 'react'
import renderer from 'react-test-renderer'

import Loading from "../index"

describe('Loading', () => {
    test('renders loading', () => {
        const component = renderer.create(
            <Loading />
        )
        let tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
    test('renders small loading', () => {
        const component = renderer.create(
            <Loading small />
        )
        let tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})