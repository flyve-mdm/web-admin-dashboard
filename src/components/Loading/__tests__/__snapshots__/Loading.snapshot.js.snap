// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders loading 1`] = `
<div
  className="loading"
  style={
    Object {
      "marginTop": -0,
    }
  }
>
  <div>
    <div
      className="loader"
    >
      <div
        className="circle"
      />
      <div
        className="circle"
      />
      <div
        className="circle"
      />
      <div
        className="circle"
      />
      <div
        className="circle"
      />
    </div>
    <p>
      
    </p>
  </div>
</div>
`;

exports[`Loading renders small loading 1`] = `
<div
  className="loading loading--small"
  style={Object {}}
>
  <div>
     
    <div
      className="loader"
    >
      <div
        className="circle"
      />
      <div
        className="circle"
      />
      <div
        className="circle"
      />
      <div
        className="circle"
      />
      <div
        className="circle"
      />
    </div>
     
  </div>
</div>
`;
